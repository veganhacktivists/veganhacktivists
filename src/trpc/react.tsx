'use client';

import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { loggerLink, unstable_httpBatchStreamLink } from '@trpc/client';
import { createTRPCReact } from '@trpc/react-query';
import { type inferRouterInputs, type inferRouterOutputs } from '@trpc/server';
import { useState } from 'react';
import superjson from 'superjson';
import { Prisma } from '@prisma/client';

import type { AppRouter } from 'server/api/root';

const createQueryClient = () => new QueryClient();

let clientQueryClientSingleton: QueryClient | undefined = undefined;
const getQueryClient = () => {
  if (typeof window === 'undefined') {
    // Server: always make a new query client
    return createQueryClient();
  }
  // Browser: use singleton pattern to keep the same query client
  return (clientQueryClientSingleton ??= createQueryClient());
};

export const api = createTRPCReact<AppRouter>();

/**
 * Inference helper for inputs.
 *
 * @example type HelloInput = RouterInputs['example']['hello']
 */
export type RouterInputs = inferRouterInputs<AppRouter>;

/**
 * Inference helper for outputs.
 *
 * @example type HelloOutput = RouterOutputs['example']['hello']
 */
export type RouterOutputs = inferRouterOutputs<AppRouter>;

superjson.registerCustom<Prisma.Decimal, string>(
  {
    isApplicable: (v): v is Prisma.Decimal => Prisma.Decimal.isDecimal(v),
    serialize: (v) => v.toJSON(),
    deserialize: (v) => new Prisma.Decimal(v),
  },
  'decimal.js',
);

export const TRPCReactProvider = (props: { children: React.ReactNode }) => {
  const queryClient = getQueryClient();

  const [trpcClient] = useState(() =>
    api.createClient({
      links: [
        loggerLink({
          enabled: (op) =>
            process.env.NODE_ENV === 'development' ||
            (op.direction === 'down' && op.result instanceof Error),
        }),
        unstable_httpBatchStreamLink({
          transformer: superjson,
          url: getBaseUrl() + '/api/trpc',
          headers: () => {
            return { 'x-trpc-source': 'nextjs-react' };
          },
        }),
      ],
    }),
  );

  return (
    <QueryClientProvider client={queryClient}>
      <api.Provider client={trpcClient} queryClient={queryClient}>
        {props.children}
      </api.Provider>
    </QueryClientProvider>
  );
};

function getBaseUrl() {
  if (typeof window !== 'undefined') return window.location.origin;
  if (process.env.URL) return `https://${process.env.URL}`;
  return `http://localhost:${process.env.PORT ?? 3000}`;
}
