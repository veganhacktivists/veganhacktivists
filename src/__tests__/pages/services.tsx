import React from 'react';
import { render } from '@testing-library/react';

import Services from '../../pages/[locale]/services';

it('should render correctly', () => {
  const { asFragment } = render(<Services />);
  expect(asFragment()).toMatchSnapshot();
});
