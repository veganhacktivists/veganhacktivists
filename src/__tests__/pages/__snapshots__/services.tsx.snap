// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<DocumentFragment>
  <div>
    <div
      class="relative p-10 md:p-32 bg-scroll bg-cover bg-no-repeat flex justify-start xl:h-screen-60% bg-center"
    >
      <div
        style="display: block; overflow: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; margin: 0px;"
      >
        <img
          alt=""
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Ffoo.png&w=3840&q=75"
          srcset="/_next/image?url=%2Ffoo.png&w=640&q=75 640w, /_next/image?url=%2Ffoo.png&w=750&q=75 750w, /_next/image?url=%2Ffoo.png&w=828&q=75 828w, /_next/image?url=%2Ffoo.png&w=1080&q=75 1080w, /_next/image?url=%2Ffoo.png&w=1200&q=75 1200w, /_next/image?url=%2Ffoo.png&w=1920&q=75 1920w, /_next/image?url=%2Ffoo.png&w=2048&q=75 2048w, /_next/image?url=%2Ffoo.png&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
        />
      </div>
      <div
        class="flex flex-col justify-center w-1/2 z-10"
      >
        <div
          class="md:p-12"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMzI1IiBoZWlnaHQ9IjMyNSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
              />
            </div>
            <img
              alt="Compassion, Creativity, Code"
              decoding="async"
              src="/_next/image?url=%2Ffoo.png&w=750&q=100"
              srcset="/_next/image?url=%2Ffoo.png&w=384&q=100 1x, /_next/image?url=%2Ffoo.png&w=750&q=100 2x"
              style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </div>
      </div>
      <div
        class="absolute inset-0 overflow-hidden"
      >
        <svg
          class="text-white overflow-visible absolute top-0 right-0 transform translate-x-1/4 -translate-y-2/4 lg:-translate-y-3/4
    "
          height="60vw"
          width="60vw"
        >
          <circle
            class="stroke-current"
            cx="30vw"
            cy="30vw"
            fill-opacity="0"
            opacity="0.1"
            r="30vw"
            stroke-width="20px"
          />
        </svg>
        <svg
          class="text-white overflow-visible absolute bottom-0 left-0 transform -translate-x-1/4 translate-y-2/4 lg:translate-y-3/4
    "
          height="68vw"
          width="68vw"
        >
          <circle
            class="stroke-current"
            cx="34vw"
            cy="34vw"
            fill-opacity="0"
            opacity="0.5"
            r="34vw"
            stroke-width="20px"
          />
        </svg>
      </div>
    </div>
    <div
      class="relative overflow-visible hidden md:block"
    >
      <div
        class="absolute left-0 bottom-0 bg-white w-16 h-16 opacity-100"
      />
      <div
        class="absolute left-0 top-0 bg-yellow w-16 h-16 opacity-100"
      />
      <div
        class="absolute left-16 bottom-0 bg-green w-32 h-32 opacity-100"
      />
      <div
        class="absolute right-0 -top-16 bg-red w-32 h-32 opacity-100"
      />
      <div
        class="absolute right-32 bottom-16 bg-orange w-16 h-16 opacity-100"
      />
      <div
        class="absolute right-32 bottom-0 bg-white w-16 h-16 opacity-100"
      />
    </div>
    <div
      class="text-grey content-center mx-auto my-32 md:w-1/2 drop-shadow-2xl text-2xl"
    >
      <h1
        class="mb-16"
      >
        <span
          class="font-italic text-3xl"
        >
          Our 
        </span>
        <b
          class="font-mono text-5xl uppercase"
        >
          services
        </b>
      </h1>
    </div>
    <div
      class="flex flex-col space-y-20 items-center mx-auto drop-shadow-2xl text-2xl mb-20"
    >
      <div>
        <div
          class="flex w-full xl:w-3/5 m-auto group min-height"
        >
          <div
            class="flex-none hidden lg:block bg-magenta"
          >
            <div
              class="transition duration-150 opacity-0 group-hover:opacity-100"
            >
              <div
                class="relative overflow-visible"
              >
                <div
                  class="absolute right-0 top-0 bg-red w-16 h-16 opacity-100"
                />
              </div>
            </div>
            <div
              class="p-10"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="Websites"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </div>
          <div
            class="text-left p-4 bg-gray-background min-height"
          >
            <h1
              class="text-4xl font-bold"
            >
              Websites
            </h1>
            <p
              class="my-3"
            >
              If you're a small vegan activist, organization, or business with little to no funding, you're the ones we want to help succeed the most! In many cases we support folks that either can't pay for website design, development or have little knowledge or time to build it themselves. We'll also take care of hosting using a dynamic dedicated server w/ Cloudflare protection and domain privacy
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="flex w-full xl:w-3/5 m-auto group min-height flex-row-reverse"
        >
          <div
            class="flex-none hidden lg:block bg-green"
          >
            <div
              class="transition duration-150 opacity-0 group-hover:opacity-100"
            >
              <div
                class="relative overflow-visible"
              >
                <div
                  class="absolute right-0 top-0 bg-green-dark w-16 h-16 opacity-100"
                />
              </div>
            </div>
            <div
              class="p-10"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="Projects"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </div>
          <div
            class="text-left p-4 bg-gray-background min-height"
          >
            <h1
              class="text-4xl font-bold"
            >
              Projects
            </h1>
            <p
              class="my-3"
            >
              Do you have an idea that would help bring us closer to a vegan world? A new website? App? Book? Movie? Song? Activism? Event? Content? Let us know, we'll help get you the resources you need to get your idea off the ground, or if we love it, we'll help you build it using our resources and team
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="flex w-full xl:w-3/5 m-auto group min-height"
        >
          <div
            class="flex-none hidden lg:block bg-yellow"
          >
            <div
              class="transition duration-150 opacity-0 group-hover:opacity-100"
            >
              <div
                class="relative overflow-visible"
              >
                <div
                  class="absolute right-0 top-0 bg-yellow-dark w-16 h-16 opacity-100"
                />
              </div>
            </div>
            <div
              class="p-10"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="Funding"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </div>
          <div
            class="text-left p-4 bg-gray-background min-height"
          >
            <h1
              class="text-4xl font-bold"
            >
              Funding
            </h1>
            <p
              class="my-3"
            >
              We're lucky enough to have a generous benefactor that allows us to fund vegan activists and small grassroot vegan organizations! Looking to start a cube of truth in your area? Interested in other types of street activism but need funding for vegan cupcakes, stickers, signs? What about support for camera gear, graphic design, development or transportation? Request funding!
            </p>
          </div>
        </div>
      </div>
      <div>
        <div
          class="flex w-full xl:w-3/5 m-auto group min-height flex-row-reverse"
        >
          <div
            class="flex-none hidden lg:block bg-orange"
          >
            <div
              class="transition duration-150 opacity-0 group-hover:opacity-100"
            >
              <div
                class="relative overflow-visible"
              >
                <div
                  class="absolute right-0 top-0 bg-orange-dark w-16 h-16 opacity-100"
                />
              </div>
            </div>
            <div
              class="p-10"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iNTAwIiBoZWlnaHQ9IjUwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="Advice"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </div>
          <div
            class="text-left p-4 bg-gray-background min-height"
          >
            <h1
              class="text-4xl font-bold"
            >
              Advice
            </h1>
            <p
              class="my-3"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vitae elit turpis. Nunc rhoncus nec sem nec tempor. Sed id finibus enim. Sed sollicitudin odio volutpat, fermentum nibh sed, vestibulum neque. Ut eu lectus at nibh consequat aliquam fermentum id velit.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
